src/main/java/mate/academy/myJdbc/dao/DeveloperDaoImpl.java

```diff
@@ -0,0 +1,240 @@
+                    "JOIN skills ON (skills.skill_id = developers_skills.skill_id)\n" +
+                    "WHERE skills.skill_branch = 'Java';");
+            ResultSet rs = statement.executeQuery();
+            Set<Developer> set = new HashSet<>();
+
+            while (rs.next()) {
+                Developer dev = getDeveloper(rs);
+                set.add(dev);
+            }
+            return set;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public Set<Developer> getAllMiddleDevs() {
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT developers.developer_id, developers.developer_name, developers.developer_age, developers.developer_salary, skills.skill_level\n" +
+                    "FROM developers\n" +
+                    "JOIN developers_skills ON (developers.developer_id = developers_skills.developer_id)\n" +
+                    "JOIN skills ON (skills.skill_id = developers_skills.skill_id)\n" +
+                    "WHERE skills.skill_level = 'Middle';");
+            ResultSet rs = statement.executeQuery();
+            Set<Developer> set = new HashSet<>();
+
+            while (rs.next()) {
+                Developer dev = getDeveloper(rs);
+                set.add(dev);
+            }
+            return set;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public int addNewDeveloper(Developer developer) {
+        try {
+            //insert dev
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO developers(developer_name, developer_age, developer_salary) VALUES(?, ?, ?)");
+            statement.setString(1, developer.getName());
+            statement.setInt(2, developer.getAge());
+            statement.setInt(3, developer.getSalary());
+            statement.execute();
+
+            //get id of developer
+            statement = connection.prepareStatement("SELECT MAX(developer_id) FROM developers;");
+            ResultSet rs = statement.executeQuery();
+            rs.next();
+            int lastDevId = rs.getInt("MAX(developer_id)");
+
+            return lastDevId;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    public Developer findDeveloper(int id) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT developer_id, developer_name, developer_age" +
+                    ", developer_salary FROM developers WHERE developer_id=?;");
+            statement.setInt(1, id);
+            ResultSet rs = statement.executeQuery();
+            rs.next();
+            Developer developer = getDeveloper(rs);
+
+            //add skills to developer
+            statement = connection.prepareStatement("SELECT skills.skill_id, skills.skill_branch, skills.skill_level, developers.developer_id\n" +
```

Comment:
I would recommend you to move this code to separate method getSkillsForDeveloper(Integer developerId) int SkillsDao